aws/src/main/python/runPilates/lambda_function.py

```diff
@@ -0,0 +1,318 @@
+                  'f1.2xlarge', 'f1.16xlarge',
+                  'g2.2xlarge', 'g2.8xlarge',
+                  'g3.4xlarge', 'g3.8xlarge', 'g3.16xlarge',
+                  'p2.xlarge', 'p2.8xlarge', 'p2.16xlarge',
+                  'p3.2xlarge', 'p3.8xlarge', 'p3.16xlarge',
+                  'r4.large', 'r4.xlarge', 'r4.2xlarge', 'r4.4xlarge', 'r4.8xlarge', 'r4.16xlarge',
+                  'r3.large', 'r3.xlarge', 'r3.2xlarge', 'r3.4xlarge', 'r3.8xlarge',
+                  'x1.16xlarge', 'x1.32xlarge',
+                  'x1e.xlarge', 'x1e.2xlarge', 'x1e.4xlarge', 'x1e.8xlarge', 'x1e.16xlarge', 'x1e.32xlarge',
+                  'd2.xlarge', 'd2.2xlarge', 'd2.4xlarge', 'd2.8xlarge',
+                  'i2.xlarge', 'i2.2xlarge', 'i2.4xlarge', 'i2.8xlarge',
+                  'h1.2xlarge', 'h1.4xlarge', 'h1.8xlarge', 'h1.16xlarge',
+                  'i3.large', 'i3.xlarge', 'i3.2xlarge', 'i3.4xlarge', 'i3.8xlarge', 'i3.16xlarge', 'i3.metal',
+                  'c5.large', 'c5.xlarge', 'c5.2xlarge', 'c5.4xlarge', 'c5.9xlarge', 'c5.18xlarge',
+                  'c5d.large', 'c5d.xlarge', 'c5d.2xlarge', 'c5d.4xlarge', 'c5d.9xlarge', 'c5d.18xlarge',
+                  'r5.large', 'r5.xlarge', 'r5.2xlarge', 'r5.4xlarge', 'r5.12xlarge', 'r5.24xlarge',
+                  'r5d.large', 'r5d.xlarge', 'r5d.2xlarge', 'r5d.4xlarge', 'r5d.12xlarge', 'r5d.24xlarge',
+                  'm5d.large', 'm5d.xlarge', 'm5d.2xlarge', 'm5d.4xlarge', 'm5d.12xlarge', 'm5d.24xlarge',
+                  'z1d.large', 'z1d.xlarge', 'z1d.2xlarge', 'z1d.3xlarge', 'z1d.6xlarge', 'z1d.12xlarge']
+
+regions = ['us-east-1', 'us-east-2', 'us-west-2']
+shutdown_behaviours = ['stop', 'terminate']
+
+s3 = boto3.client('s3')
+ec2 = None
+
+
+def init_ec2(region):
+    global ec2
+    ec2 = boto3.client('ec2', region_name=region)
+
+
+def validate(name):
+    return True
+
+
+def deploy(script, instance_type, region_prefix, shutdown_behaviour, instance_name, volume_size):
+    res = ec2.run_instances(BlockDeviceMappings=[
+        {
+            'DeviceName': '/dev/sda1',
+            'Ebs': {
+                'VolumeSize': volume_size,
+                'VolumeType': 'gp2'
+            }
+        }
+    ],
+        ImageId=os.environ[region_prefix + 'IMAGE_ID'],
+        InstanceType=instance_type,
+        UserData=script,
+        KeyName=os.environ[region_prefix + 'KEY_NAME'],
+        MinCount=1,
+        MaxCount=1,
+        SecurityGroupIds=[os.environ[region_prefix + 'SECURITY_GROUP']],
+        IamInstanceProfile={'Name': os.environ['IAM_ROLE']},
+        InstanceInitiatedShutdownBehavior=shutdown_behaviour,
+        TagSpecifications=[{
+            'ResourceType': 'instance',
+            'Tags': [{
+                'Key': 'Name',
+                'Value': instance_name
+            }]
+        }])
+    return res['Instances'][0]['InstanceId']
+
+
+def get_dns(instance_id):
+    host = None
+    while host is None:
```

Comment:
Maybe this should have some form of max tries in case the instance failed for some reason - otherwise this will spin forever. Or an exponential backoff?

Reply:
good point, will add something like max_tries